lib/src/taiko/protocol_instance.rs

```diff
@@ -1,17 +1,36 @@
 use anyhow::Result;
+use ethers_core::{
+    abi::{encode_packed, Token},
+    types::U256 as EthU256,
+};
 use zeth_primitives::{
     block::Header,
-    ethers::{from_ethers_h256, from_ethers_u256},
+    ethers::from_ethers_h256,
     keccak,
     taiko::{
         deposits_hash, string_to_bytes32, BlockMetadata, EthDeposit, ProtocolInstance, Transition,
         ANCHOR_GAS_LIMIT,
     },
-    TxHash, U256,
+    TxHash,
 };
+use alloy_sol_types::SolValue;
 
 use crate::taiko::host::TaikoExtra;
 
+
+fn calc_difficulty(prevrando: EthU256, num_blocks: u64, block_num: EthU256) -> [u8; 32] {
+    // meta.difficulty = keccak256(abi.encodePacked(block.prevrandao, b.numBlocks, block.number));
+    let prevrando_bytes: [u8; 32] = prevrando.into();
```

Comment:
No need to transfers between types because `SolValue` has implemented for primitive types and alloy types

Reply:
Not this one, it's a quick test before I switch back to zeth data type. Just submitted.